File: ruby/algo_ladder/show_me_the_money/show_me_the_money_explained.rb
# Given a string, return true if the “$” character is contained within the string or false if it is not.

# Input: “i hate $ but i love money i know i know im crazy”
# Output: true

# Input: “abcdefghijklmnopqrstuvwxyz”
# Output: false

# Set up function that takes in a test input_string
def show_me_the_money (input_string)
  # Create an index that we will use to loop through our string
  index = 0
  # While loop that loops through our string
  while index < input_string.length
    # If the current element being looped over is a $ then return true
    if input_string[index] === "$"
      return true
    end
    # Increment our index
    index += 1
  end
  # If true is not returned then return false
  return false
end
# Test function with our two test inputs
p show_me_the_money("i hate $ but i love money i know i know im crazy")
p show_me_the_money("abcdefghijklmnopqrstuvwxyz")

File: javascript/algoLadder/showMeTheMoney/showMeTheMoney_explained.js
// Given a string, return true if the “$” character is contained within the string or false if it is not.

// Input: “i hate $ but i love money i know i know im crazy”
// Output: true

// Input: “abcdefghijklmnopqrstuvwxyz”
// Output: false

// Using .includes method 
// Create function that takes in a string
const showMeTheMoney = function (string) {
  // If the string includes the $ symbol then return true.
  if (string.includes("$")) {
    return true
    // Else return false
  } else {
    return false
  }
}
// Call the function with the input string which will return false
console.log(showMeTheMoney("i hate $ but i love money i know i know im crazy"))
console.log(showMeTheMoney("abcdefghijklmnopqrstuvwxyz"))

// Using a while loop
// Create function that takes in a string
const showMeTheMoneyLoop = function (string) {
  // Create an index that we can use to increment our way through our string
  let index = 0
  // Create a while loop that will loop through each element in the string
  while (index < string.length) {
    // If the current element in the string is equal to the $ symbol then return true
    if (string[index] === "$") {
      return true
    }
    // Increment the index each loop
    index += 1
  }
  // Return false if none of the string elements are equal to $
  return false
}
// Call the function with the input string which will return false
console.log(showMeTheMoneyLoop("i hate $ but i love money i know i know im crazy"))
console.log(showMeTheMoneyLoop("abcdefghijklmnopqrstuvwxyz"))